pkgs/games/hmcl/default.nix

```
@@ -0,0 +1,58 @@
+{ lib, stdenv
+, fetchurl
+, jre
+, makeWrapper
+, makeDesktopItem
+, copyDesktopItems
+}:
+
+stdenv.mkDerivation rec {
+  pname = "hmcl";
+  version = "3.5.3.223";
+
+  src = ./.;
+
+  hmcl = fetchurl {
+    url = "https://github.com/huanghongxun/HMCL/releases/download/v${version}/HMCL-${version}.jar";
+    sha256 = "sha256-8g2FMvAiAKfxJUY0G7wl6d44wOpVsknFHGZ85IkOzFc=";
+  };
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  nativeBuildInputs = [ makeWrapper copyDesktopItems ];
+
+  desktopItem = makeDesktopItem {
+    name = "hmcl";
+    exec = "hmcl";
+    icon = "hmcl";
+    comment = "A Minecraft Launcher which is multi-functional, cross-platform and popular";
+    desktopName = "HMCL";
+    categories = [ "Game" ];
+  };
+
+  desktopItems = [ desktopItem ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/opt/hmcl
```

This is the standard, see the [java section](https://nixos.org/manual/nixpkgs/stable/#sec-language-java) of the nixpkgs manual.
```suggestion
    mkdir -p $out/share/java
```